scrape/target.go

```diff
@@ -420,10 +428,18 @@ func (app *maxSchemaAppender) AppendHistogram(ref storage.SeriesRef, lset labels
 	return ref, nil
 }
 
```

Comment:
Can we add some exported commentary? Probably similar to `DiscoveredLabels`

```suggestion
PopulateDiscoveredLabels builds target's labels before any processing.
```